evap/results/views.py

```diff
@@ -86,8 +86,13 @@ def course_detail(request, semester_id, course_id):
                 answers = [answer for answer in result.answers if user_can_see_text_answer(request.user, represented_users, answer, public_view)]
                 if answers:
                     results.append(TextResult(question=result.question, answers=answers))
-            else:
+            elif show_grades or isinstance(result, HeadingResult):
                 results.append(result)
+            elif isinstance(result, RatingResult):
+                results.append(RatingResult(question=result.question, total_count=result.total_count, average=None, counts=None, warning=result.warning))
+            elif isinstance(result, YesNoResult):
+                results.append(YesNoResult(question=result.question, total_count=result.total_count, average=None, counts=None, warning=result.warning, approval_count=None))
```

Comment:
previously, the template needed to remove the grades if the user was not allowed to see them, i'm missing a corresponding change in this commit...?

Reply:
the previous commit changed the distribution bar and average grade badge rendering. they are not doing checks by themselves anymore and always display data if they receive some.